build.gradle

```diff
@@ -102,28 +93,80 @@ wrapper {
 }
 
 // BuildDate function adapted from http://stackoverflow.com/questions/19172565/how-append-date-build-to-versionnamesuffix-on-gradle
-def buildDate() {
+def buildDate = {
     def df = new SimpleDateFormat('yyyy-MM-dd')
+
     df.setTimeZone(TimeZone.getTimeZone('UTC'))
+
     return df.format(new Date())
 }
 
 gitProperties {
-    customProperty 'buildDate', "${buildDate()}"
+    customProperty 'buildDate', buildDate
     customProperty 'develop', 'true'
 }
 
-task buildLinuxPackage(type: Exec) {
+def stringify(string) {
+    return '"' + string + '"';
+}
+
+def appVersion = {
+    return version
+            .replaceAll("\\-", ".")
+            .replaceAll("alpha", "0")
+            .replaceAll("beta", "0")
+            .replaceAll("release", "0")
+}
+
+task buildDebPackage(type: Exec) {
     commandLine 'jpackage',
             '--verbose',
-            '--name', 'Rails (18xx)',
-            '--description', '"Rails is a moderator program for 18xx games"',
+            '--type', 'deb',
+            '--name', stringify(appName),
+            '--description', stringify(description),
+            '--vendor', stringify(vendor),
+            '--license-file', licenceFile,
             '--icon', './native/rails.png',
             '--file-associations', './native/rails-linux-association.properties',
-            '--app-version', '2.2.0',
+            '--app-version', appVersion,
             '--input', 'build/libs',
-            '--main-jar', 'rails-2.2-beta-all.jar',
+            '--main-jar', "rails-${version}-all.jar",
```

Comment:
You should also be able to get the jar name via shadow and hence don't need to construct the name here-- little bit safer and less error prone?

Reply:
How do you get the filename from shadow?

Reply:
I believe the shadow plugin replaces the default output jar name with the name of the uber jar